transforms/transforms.go

```diff
@@ -18,6 +19,48 @@ func TableTransform(transform func(optimus.Row, chan<- optimus.Row) error) optim
 	}
 }
 
+// Explode returns a TransformFunc that takes a row that contains an array
+// in the specified key and returns a duplicate of the row for each value in the array.
+// For example, exploding on "school" for:
+// optimus.Row{"id": "a", "school": []string{"Jones Elementary", "Cedar Middle"}}
+// becomes
+// []optimus.Row{
+//     {"id": "a", "school": "Jones Elementary"},
+//     {"id": "a", "school": "Cedar Middle"},
+// }
+func Explode(keyToExplode string) optimus.TransformFunc {
+	return TableTransform(func(row optimus.Row, out chan<- optimus.Row) error {
+        objToExplode := row[keyToExplode]
+
+        switch reflect.TypeOf(objToExplode).Kind() {
```

Comment:
I didn't see anywhere else that made assumptions about the types of elements in the row, so I didn't want to start here. Also, as far as my limited understanding of Go types goes, I'm pretty sure if what was passed in was a slice of custom interfaces (like []oauth.authTypes), it would not match []interface{}.